Fabric.Authorization.Domain/Services/GroupService.cs

```diff
@@ -228,13 +243,21 @@ public async Task<Group> AddChildGroups(Group group, IEnumerable<string> childGr
                 throw new AlreadyExistsException<Group>($"The following groups are already children of group {group.Name}: {string.Join(", ", existingAssociations.Select(g => g.Name))}");
             }
 
-            return await _groupStore.AddChildGroups(group, childGroups);
+            var persistedGroup = await _groupStore.AddChildGroups(group, childGroups);
+            await this._roleManager.RefreshDosAdminRolesAsync(group.Children.SelectMany(childGroup => childGroup.Users));
```

Comment:
Do we need this in the child groups logic? This is just removing the association between a parent and child group.

Reply:
Ahh I see, was under the impression that the child group could be a new group, but looking at it now it takes in names rather than an actuall group

Reply:
How do the roles relate to users here, for example, if a user is in the parent group and a child group is added, do the users from the parent group inherit the child group roles as well?

Reply:
Yes, users will inherit the permissions of the roles tied to the child groups. Users can only be added to `custom` groups, and only `custom` groups can be parents,, while only `directory` (i.e., AD) groups an be children. So perhaps this (and the sync logic in `RemoveChildGroups`) does need to stay now that I think about it. However, I don't think the logic in the query is correct:

1) `ChildGroup` navigation collection does not include `Users`, but this is irrelevant since AD groups cannot have users.
2) I think the `IsUserASuperAdmin` method needs to be changed to check the group's children's roles for the 2 roles we want to sync.
3) This call should then just be:
`await this._roleManager.RefreshDosAdminRolesAsync(group.Users);`